lib/Search/UnifiedSearchFilterPlugin.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+declare(strict_types=1);
+/**
+ * @copyright Copyright (c) 2024 Fon E. Noel NFEBE <me@nfebe.com>
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace OCA\Talk\Search;
+
+use OCP\AppFramework\Http\Events\BeforeTemplateRenderedEvent;
+use OCP\EventDispatcher\Event;
+use OCP\EventDispatcher\IEventListener;
+use OCP\IRequest;
+use OCP\Util;
+
+/**
+ * @template-implements IEventListener<Event>
+ */
+class UnifiedSearchFilterPlugin implements IEventListener {
+
+	public function __construct(
+		private IRequest $request,
+	) {
+	}
+
+	public function handle(Event $event): void {
+		if (!($event instanceof BeforeTemplateRenderedEvent)) {
+			return;
+		}
+
+		if (!$event->isLoggedIn()) {
+			return;
+		}
+
+		// REVIEW: Only add the talk filter if user is on talk app?
+		if (str_starts_with($this->request->getPathInfo(), '/apps/spreed')) {
+			Util::addScript('spreed', 'talk-search');
```

Comment:
Script is around 460 kB dev / 270 kB production. Can't see nothing in the bundle analyzer, that we could get rid of (only minor optimizations from our side, maybe)

Reply:
Would be nice to have it lazy-loaded, when opening unified search modal

Reply:
Not sure whether this a global backend event for such lazy load to depend on right now.